File: python3/p10.py
#!/usr/bin/python3

import sys
import math

def main():
    limit = 2000000
    if len(sys.argv) > 1:
        limit = int(sys.argv[1])
    cross_limit = math.floor(math.sqrt(limit))
    sieve = [False] * limit
    for n in range(4, limit, 2):
        sieve[n] = True
    for n in range(3, cross_limit + 1, 2):
        if not sieve[n]:
            for m in range(n*n, limit, 2*n):
                sieve[m] = True
    sum = 0
    for n in range(2, limit):
        if not sieve[n]:
            sum += n
    print(int(sum))

if __name__ == "__main__":
    main()

File: javascript/p10.js
#!/usr/bin/env node

let limit = 2000000;
if (process.argv.length > 2) {
  limit = process.argv[2];
}
let cross_limit = Math.floor(Math.sqrt(limit));
let sieve = new Array(2000000).fill(false);
for(n = 4; n < limit; n += 2){
  sieve[n] = true;
}
for (n = 3; n < cross_limit; n += 2) {
  if (!sieve[n]) {
    for (m = n * n; m < limit; m += (2 * n)) {
      sieve[m] = true;
    }
  }
}
let sum = 0;
for (n = 2; n < limit; n++) {
  if (!sieve[n]) {
    sum += n;
  }
}
console.log(sum);
